src/olympia/users/models.py

```diff
@@ -499,23 +493,11 @@ def set_password(self, password):
     def check_password(self, password):
         raise NotImplementedError('cannot check password')
 
-    def log_login_attempt(self, successful):
-        """Log a user's login attempt"""
-        self.last_login_attempt = datetime.now()
-        self.last_login_attempt_ip = core.get_remote_addr()
-
-        if successful:
-            log.debug(u"User (%s) logged in successfully" % self)
-            self.failed_login_attempts = 0
-            self.last_login_ip = core.get_remote_addr()
-        else:
-            log.debug(u"User (%s) failed to log in" % self)
-            if self.failed_login_attempts < 16777216:
-                self.failed_login_attempts += 1
-
-        self.save(update_fields=['last_login_ip', 'last_login_attempt',
-                                 'last_login_attempt_ip',
-                                 'failed_login_attempts'])
+    @staticmethod
+    def user_logged_in(sender, request, user, **kwargs):
+        """Log when a user logs in and records its IP address."""
+        log.debug(u'User (%s) logged in successfully' % user)
+        user.update(last_login_ip=core.get_remote_addr() or '')
```

Comment:
Do we _want_ to wipe the ip address if there isn't one for the current login?  (How would that happen?)    On the other hand, keeping old ip addresses isn't correct either.

Reply:
I'm not too sure either, I figured it was the most correct way of handling this.

It shouldn't happen... unless somehow `REMOTE_ADDR` was empty, or we're outside the normal request/response cycle. It does happen in tests for this reason. Since the field does not accept `NULL` I had to do this.